src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -0,0 +1,529 @@
+                3 => WebSocketState.Closed, // 3 (CLOSED)
+                _ => WebSocketState.None
+            };
+
+        public override WebSocketCloseStatus? CloseStatus => _innerWebSocket == null ? null : _innerWebSocketCloseStatus;
+
+        public override string? CloseStatusDescription => _innerWebSocket == null ? null : _innerWebSocketCloseStatusDescription;
+
+        public override string? SubProtocol => _innerWebSocket != null && !_innerWebSocket.IsDisposed ? _innerWebSocket!.GetObjectProperty("protocol")?.ToString() : null;
+
+        #endregion Properties
+
+        internal async Task ConnectAsyncJavaScript(Uri uri, CancellationToken cancellationToken, List<string>? requestedSubProtocols)
+        {
+            var tcsConnect = new TaskCompletionSource<bool>();
+
+            // For Abort/Dispose.  Calling Abort on the request at any point will close the connection.
+            _cts.Token.Register(AbortRequest);
+
+            // Wrap the cancellationToken in a using so that it can be disposed of whether
+            // we successfully connected or failed trying.
+            // Otherwise any timeout/cancellation would apply to the full session.
+            // In the failure case we need to release the references and dispose of the objects.
+            using (cancellationToken.Register(() => tcsConnect.TrySetCanceled()))
+            {
+                try
+                {
+                    JavaScript.Array? subProtocols = null;
+                    if (requestedSubProtocols?.Count > 0)
+                    {
+                        subProtocols = new JavaScript.Array();
+                        foreach (var item in requestedSubProtocols)
+                        {
+                            subProtocols.Push(item);
+                        }
+                    }
+                    _innerWebSocket = subProtocols == null ? new HostObject("WebSocket", uri.ToString()) : new HostObject("WebSocket", uri.ToString(), subProtocols);
+                    _innerWebSocket.SetObjectProperty("binaryType", "arraybuffer");
+
+                    subProtocols?.Dispose();
+
+                    // Setup the onError callback
+                    _onError = new Action<JSObject>((errorEvt) =>
+                    {
+                        errorEvt.Dispose();
+                    });
+
+                    // Attach the onError callback
+                    _innerWebSocket.SetObjectProperty("onerror", _onError);
+
+                    // Setup the onClose callback
+                    _onClose = new Action<JSObject>((closeEvt) =>
+                    {
+                        _innerWebSocketCloseStatus = (WebSocketCloseStatus)closeEvt.GetObjectProperty("code");
+                        _innerWebSocketCloseStatusDescription = closeEvt.GetObjectProperty("reason")?.ToString();
+                        var mess = new ReceivePayload(ArraySegment<byte>.Empty, WebSocketMessageType.Close);
```

Comment:
- Use EmptyArray instead of ArraySegment<byte>.Empty
- change local variable name

Reply:
Addressed